diff --git a/LoopKit/DosingDecisionStore.swift b/LoopKit/DosingDecisionStore.swift
index c1059354..7d80fa61 100644
--- a/LoopKit/DosingDecisionStore.swift
+++ b/LoopKit/DosingDecisionStore.swift
@@ -317,6 +317,9 @@ public struct StoredDosingDecision: DosingDecision {
     public var warnings: [Issue]
     public var errors: [Issue]
     public var syncIdentifier: UUID
+    //wp begin
+    public var automaticDosingStrategy: String
+    //wp end
 
     public init(id: UUID = UUID(),
                 date: Date = Date(),
@@ -344,7 +347,10 @@ public struct StoredDosingDecision: DosingDecision {
                 enactedBolusAmount: Double? = nil,
                 warnings: [Issue] = [],
                 errors: [Issue] = [],
-                syncIdentifier: UUID = UUID()) {
+                //wp begin
+                syncIdentifier: UUID = UUID(),
+                automaticDosingStrategy: String = "tempBasalOnly") {
+                //wp end
         self.id = id
         self.date = date
         self.controllerTimeZone = controllerTimeZone
@@ -372,6 +378,9 @@ public struct StoredDosingDecision: DosingDecision {
         self.warnings = warnings
         self.errors = errors
         self.syncIdentifier = syncIdentifier
+        //wp begin
+        self.automaticDosingStrategy = automaticDosingStrategy
+        //wp end
     }
 
     public struct Settings: Codable, Equatable {
@@ -471,7 +480,10 @@ extension StoredDosingDecision: Codable {
                   enactedBolusAmount: try container.decodeIfPresent(Double.self, forKey: .enactedBolusAmount),
                   warnings: try container.decodeIfPresent([Issue].self, forKey: .warnings) ?? [],
                   errors: try container.decodeIfPresent([Issue].self, forKey: .errors) ?? [],
-                  syncIdentifier: try container.decode(UUID.self, forKey: .syncIdentifier))
+                  //wp begin
+                  syncIdentifier: try container.decode(UUID.self, forKey: .syncIdentifier),
+                  automaticDosingStrategy: try container.decode(String.self, forKey: .automaticDosingStrategy))
+                  //wp end
     }
 
     public func encode(to encoder: Encoder) throws {
@@ -503,6 +515,9 @@ extension StoredDosingDecision: Codable {
         try container.encodeIfPresent(!warnings.isEmpty ? warnings : nil, forKey: .warnings)
         try container.encodeIfPresent(!errors.isEmpty ? errors : nil, forKey: .errors)
         try container.encode(syncIdentifier, forKey: .syncIdentifier)
+        //wp begin
+        try container.encode(automaticDosingStrategy, forKey: .automaticDosingStrategy)
+        //wp end
     }
 
     private enum CodingKeys: String, CodingKey {
@@ -533,6 +548,9 @@ extension StoredDosingDecision: Codable {
         case warnings
         case errors
         case syncIdentifier
+        //wp begin
+        case automaticDosingStrategy
+        //wp end
     }
 }
 
diff --git a/LoopKit/Resources/Localizable.xcstrings b/LoopKit/Resources/Localizable.xcstrings
index cbd583ae..c1d46adb 100644
--- a/LoopKit/Resources/Localizable.xcstrings
+++ b/LoopKit/Resources/Localizable.xcstrings
@@ -2062,6 +2062,12 @@
         }
       }
     },
+    "Basal Lock" : {
+      "comment" : "Title text for basal lock setting"
+    },
+    "Basal Lock prevents the basal rate from being throttled if the blood glucose is above a certain level." : {
+      "comment" : "Descriptive text for basal lock (1: app name)"
+    },
     "Basal Rates" : {
       "comment" : "Title text for basal rates",
       "localizations" : {
@@ -10327,6 +10333,9 @@
     },
     "Perceived Intensity" : {
 
+    },
+    "Please note that this value is outside of the recommended range." : {
+      "comment" : "Descriptive text for saving settings outside the recommended range for basal lock"
     },
     "Please re-enable sharing in Health" : {
       "comment" : "The error recovery suggestion when Health sharing was denied",
@@ -12111,6 +12120,9 @@
         }
       }
     },
+    "The value you have entered for Basal Lock is outside of the recommended range." : {
+      "comment" : "Descriptive text for guardrail outside value warning for basal lock"
+    },
     "The value you have entered is higher than what is typically recommended for most people." : {
       "comment" : "Descriptive text for guardrail high value warning",
       "localizations" : {
@@ -12254,6 +12266,9 @@
         }
       }
     },
+    "The value you have entered is higher than what is typically recommended." : {
+      "comment" : "Descriptive text for guardrail high value warning for basal lock"
+    },
     "The value you have entered is lower than what is typically recommended for most people." : {
       "comment" : "Descriptive text for guardrail low value warning",
       "localizations" : {
@@ -12397,6 +12412,9 @@
         }
       }
     },
+    "The value you have entered is lower than what is typically recommended." : {
+      "comment" : "Descriptive text for guardrail low value warning for basal lock"
+    },
     "The values you have entered are outside of what is typically recommended for most people." : {
       "comment" : "Descriptive text for guardrail high value warning",
       "extractionState" : "stale",
diff --git a/LoopKit/Service/Remote/RemoteActionDelegate.swift b/LoopKit/Service/Remote/RemoteActionDelegate.swift
index 8f824906..33ff3421 100644
--- a/LoopKit/Service/Remote/RemoteActionDelegate.swift
+++ b/LoopKit/Service/Remote/RemoteActionDelegate.swift
@@ -13,4 +13,8 @@ public protocol RemoteActionDelegate: AnyObject {
     func cancelRemoteOverride() async throws
     func deliverRemoteCarbs(amountInGrams: Double, absorptionTime: TimeInterval?, foodType: String?, startDate: Date?) async throws
     func deliverRemoteBolus(amountInUnits: Double, decisionId: UUID?) async throws
+    //wp begin
+    func deliverRemoteBolus(amountInUnits: Double, amountInGrams: Double, absorptionTime: TimeInterval?, foodType: String?, startDate: Date?) async throws
+    func enactRemoteDosingStrategy(automaticDosingStrategy: AutomaticDosingStrategy) async throws
+    //wp end
 }
diff --git a/LoopKitUI/Extensions/Environment+Authenticate.swift b/LoopKitUI/Extensions/Environment+Authenticate.swift
index 8dff88de..67327359 100644
--- a/LoopKitUI/Extensions/Environment+Authenticate.swift
+++ b/LoopKitUI/Extensions/Environment+Authenticate.swift
@@ -17,7 +17,7 @@ public struct LocalAuthentication {
     public static let deviceOwnerCheck: AuthenticationChallenge = { authenticationChallengeDescription, completion in
         let context = LAContext()
         var error: NSError?
-        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+        if false && context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
             context.evaluatePolicy(.deviceOwnerAuthentication,
                                    localizedReason: authenticationChallengeDescription,
                                    reply: { (success, error) in
diff --git a/LoopKitUI/Resources/Localizable.xcstrings b/LoopKitUI/Resources/Localizable.xcstrings
index 2c7f71b9..e0bc2e6b 100644
--- a/LoopKitUI/Resources/Localizable.xcstrings
+++ b/LoopKitUI/Resources/Localizable.xcstrings
@@ -4241,6 +4241,9 @@
           }
         }
       }
+    },
+    "A checkmark next to a profile name indicates that it's the active one.\n\nUse ‘+’ to create a new profile capturing your glucose targets, carb ratios, basal rates, and insulin sensitivities.\n\nTap on any profile to review its settings in detail. In detail view, you can Load, Rename or Delete that particular profile.\n\nWant to change the order of profiles? Simply press and hold on a profile, then drag it to your desired position." : {
+
     },
     "A Correction Range is different. The Correction Range is a " : {
 
@@ -4391,6 +4394,9 @@
     },
     "A percentage " : {
 
+    },
+    "A profile with this name already exists. Would you like to overwrite it?" : {
+
     },
     "A value of 0 U/hr means you will be scheduled to receive no basal insulin." : {
       "comment" : "Warning text for basal rate of 0 U/hr",
@@ -5889,6 +5895,9 @@
           }
         }
       }
+    },
+    "Are you sure you want to delete the profile %@? This action cannot be undone." : {
+
     },
     "Are you sure you want to delete this preset?" : {
 
@@ -6042,6 +6051,9 @@
         }
       }
     },
+    "Authentication is required to save this setting." : {
+      "comment" : "Authentication challenge description for basal lock threshold"
+    },
     "Back" : {
       "comment" : "Back navigation button title",
       "localizations" : {
@@ -6191,6 +6203,12 @@
         }
       }
     },
+    "Basal Lock prevents the basal rate from being throttled if the blood glucose is above a certain level." : {
+      "comment" : "Description for basal lock threshold setting"
+    },
+    "Basal Lock Threshold" : {
+      "comment" : "Title for basal lock threshold setting"
+    },
     "Basal Rates" : {
       "comment" : "Title text for multi-value basal rate warning",
       "localizations" : {
@@ -6795,6 +6813,9 @@
     },
     "Before deciding to adjust your overall insulin," : {
 
+    },
+    "Before proceeding, review the profile details below. Scroll to find options to Load, Rename, or Delete." : {
+
     },
     "below 100%" : {
 
@@ -10849,6 +10870,9 @@
     },
     "Delete Preset" : {
 
+    },
+    "Delete Profile" : {
+
     },
     "Delivery Limits" : {
       "comment" : "Title text for crossed thresholds guardrail warning",
@@ -11290,6 +11314,9 @@
           }
         }
       }
+    },
+    "Do you want to load the profile %@?" : {
+
     },
     "Done" : {
       "comment" : "Text for dismiss button\nText for done button\nVideo player done button label",
@@ -12458,6 +12485,9 @@
           }
         }
       }
+    },
+    "Enable Basal Lock" : {
+
     },
     "Enable Indefinitely" : {
       "comment" : "The text for the indefinite custom preset duration setting",
@@ -13342,6 +13372,9 @@
           }
         }
       }
+    },
+    "Error loading profile" : {
+
     },
     "Event History" : {
       "comment" : "Segmented button title for insulin delivery log event history",
@@ -14512,6 +14545,9 @@
           }
         }
       }
+    },
+    "Got it!" : {
+
     },
     "High Basal Rate" : {
       "comment" : "Title text for the high basal rate warning",
@@ -16389,6 +16425,9 @@
           }
         }
       }
+    },
+    "Information" : {
+
     },
     "Insulin adjustment is above the safety threshold" : {
 
@@ -17136,6 +17175,12 @@
           }
         }
       }
+    },
+    "Load" : {
+
+    },
+    "Load Profile" : {
+
     },
     "Low Basal Rate" : {
       "comment" : "Title text for the low basal rate warning",
@@ -20225,6 +20270,9 @@
           }
         }
       }
+    },
+    "New Profile" : {
+
     },
     "Next Date" : {
 
@@ -20525,6 +20573,12 @@
     },
     "of scheduled" : {
 
+    },
+    "Off" : {
+
+    },
+    "OK" : {
+
     },
     "or" : {
 
@@ -21425,6 +21479,12 @@
           }
         }
       }
+    },
+    "Overwrite" : {
+
+    },
+    "Overwrite Existing Profile" : {
+
     },
     "Pre-Meal" : {
       "comment" : "Title for the pre-meal override range",
@@ -22012,6 +22072,15 @@
         }
       }
     },
+    "Profile Name" : {
+
+    },
+    "Profile Outdated" : {
+
+    },
+    "Profiles" : {
+      "comment" : "Title on ProfileView"
+    },
     "Progressing." : {
       "comment" : "Accessibility label for ProgressIndicatorView when showIndeterminantProgress",
       "localizations" : {
@@ -22748,6 +22817,12 @@
     },
     "Recommended starting values" : {
 
+    },
+    "Rename" : {
+
+    },
+    "Rename Profile" : {
+
     },
     "Repeat" : {
 
@@ -23947,6 +24022,9 @@
     },
     "Save and Schedule for Later" : {
 
+    },
+    "Save Basal Lock Threshold?" : {
+      "comment" : "Alert title for confirming a basal lock threshold outside the recommended range"
     },
     "Save Basal Rates?" : {
       "comment" : "Alert title for confirming basal rates outside the recommended range",
@@ -28510,6 +28588,9 @@
           }
         }
       }
+    },
+    "The current therapy settings have changed. Do you want to update the profile named: %@?" : {
+
     },
     "The currently selected fast acting insulin model will be used as a default." : {
       "comment" : "Description for selection when no insulin type is selected.",
@@ -32441,6 +32522,12 @@
     },
     "Until I turn off" : {
 
+    },
+    "Update" : {
+
+    },
+    "Use ‘+’ to create a new profile capturing your glucose targets, carb ratios, basal rates, and insulin sensitivities." : {
+
     },
     "Use Scheduled Range" : {
 
@@ -33780,6 +33867,9 @@
           }
         }
       }
+    },
+    "Yes" : {
+
     },
     "Yes, Delete" : {
 
@@ -35517,6 +35607,9 @@
           }
         }
       }
+    },
+    "Your setting for basal lock is outside the recommended range." : {
+      "comment" : "Descriptive text for saving settings outside the recommended range for basal lock"
     }
   },
   "version" : "1.1"
diff --git a/LoopKitUI/Views/InsulinSensitivityScalingTableViewCell.swift b/LoopKitUI/Views/InsulinSensitivityScalingTableViewCell.swift
index 250f71b7..442ae5df 100644
--- a/LoopKitUI/Views/InsulinSensitivityScalingTableViewCell.swift
+++ b/LoopKitUI/Views/InsulinSensitivityScalingTableViewCell.swift
@@ -16,7 +16,7 @@ protocol InsulinSensitivityScalingTableViewCellDelegate: AnyObject {
 
 final class InsulinSensitivityScalingTableViewCell: UITableViewCell {
 
-    private let allScaleFactorPercentages = Array(stride(from: 10, through: 200, by: 10))
+    private let allScaleFactorPercentages = Array(stride(from: 50, through: 200, by: 5))
 
     @IBOutlet private weak var titleLabel: UILabel!
     
