diff --git a/Localization/Localizable.xcstrings b/Localization/Localizable.xcstrings
index e820777..8f85e18 100644
--- a/Localization/Localizable.xcstrings
+++ b/Localization/Localizable.xcstrings
@@ -64780,7 +64780,7 @@
         "zh-Hans" : {
           "stringUnit" : {
             "state" : "translated",
-            "value" : "开关泵类型"
+            "value" : "切换泵类型"
           }
         }
       }
