diff --git a/Localization/Localizable.xcstrings b/Localization/Localizable.xcstrings
index 8056830..93abfc2 100644
--- a/Localization/Localizable.xcstrings
+++ b/Localization/Localizable.xcstrings
@@ -66007,7 +66007,7 @@
         "zh-Hans" : {
           "stringUnit" : {
             "state" : "translated",
-            "value" : "开关泵类型"
+            "value" : "切换泵类型"
           }
         }
       }
