diff --git a/Localization/Localizable.xcstrings b/Localization/Localizable.xcstrings
index 72da0f4..83a1bab 100644
--- a/Localization/Localizable.xcstrings
+++ b/Localization/Localizable.xcstrings
@@ -66077,7 +66077,7 @@
         "zh-Hans" : {
           "stringUnit" : {
             "state" : "translated",
-            "value" : "开关泵类型"
+            "value" : "切换泵类型"
           }
         }
       }
